### DISCUSSION TITLE
Failed to start Saleor Storefront and Dashboard

### DISCUSSION THREAD

**Discussion by knimer** [saleor/saleor#9066] [Deployment]
Hello,

I created a Digital Ocean droplet with Docker Image. Which means Docker and Docker Compose are completely installed. Here are the details:

**OS**: Ubuntu 20.04
**Droplet Specs:** 4 GB Memory / 2 Intel vCPUs / 80 GB Disk
**Docker version** 20.10.7, build f0df350
**Docker-compose** version 1.27.4, build 40524192

--
- Configures Docker per the [official Docker recommendations](https://docs.docker.com/install/linux/docker-ce/ubuntu/).
- Configures Docker Compose per the [official Docker Compose recommendations](https://docs.docker.com/compose/install/#install-compose).

--
- Linux user is not root, but in **sudo** and **docker** groups.
- All setup was done using the **non-root** user

--
- Increased the **file watchers limit** to the max possible.

--

Followed the steps in following URL by the word - **copy and paste commands**:

https://docs.saleor.io/docs/3.0/developer/installation

--

All commands worked fine and successfully completed.

Except at the end when trying to run the services by the command:

`docker-compose up`

Which does not start the storefront and does not start the dashboard.

I can see this:

![image](https://user-images.githubusercontent.com/32583831/152156832-10b9dcee-dabb-42ba-9717-7ab8c61634eb.png)

Accessing the storefront and the dashboard does not work.

Sometimes, the dashboard shows only the login screen, when entering the correct credentials, nothing happens. Here is the screen:

![image](https://user-images.githubusercontent.com/32583831/152157876-f16be655-fc21-4724-85e4-45af845a0273.png)

I tried to do the setup multiple time... it simply does not work to run the services.

--

So, did I miss anything?
Is this simply a bug or a mis-configuration?

I appreciate the help

K

**Comment by karolkielecki** [comment]
Hi @knimer, is that https://github.com/saleor/saleor-platform?

**Reply by knimer** [reply]
Yes.

**Comment by knimer** [comment]
If it may help. Tried to deploy on Heorku. Got the core and dashboard running but store-front not working as well!

This time, here is Heroku's Build Log:

--
If it may help. Tried to deploy on Heorku. Got the core and dashboard running but store-front not working as well! Also, I get the same error in Docker and Heroku deployments.

The log from Heorku shows this error:

**Error: Network error: request to http://localhost:8000/graphql/ failed, reason: connect ECONNREFUSED 127.0.0.1:8000**

--

Look for this error in the last parts of the attached log.

[saleor-storefront-heroku-build.log](https://github.com/saleor/saleor/files/8001764/saleor-storefront-heroku-build.log)
.

**Reply by karolkielecki** [reply]
Saleor-platform hasn't been designed for deployment purposes, as stated in the readme section. Only for development purposes. Also, there are some issues with that repo--we'll take a look at it once we're more available.

**Reply by buynyakov** [reply]
> Saleor-platform hasn't been designed for deployment purposes, as stated in the readme section. Only for development purposes. Also, there are some issues with that repo--we'll take a look at it once we're more available.

@karolkielecki any updates on how to start frontend ?

**Comment by knimer** [comment]
Thanks for the answer.

Yet, I tried even Heroku Button and still the saleor-storefront did not work. Maybe because it is all dependent on the repos.

FYI - I also tried to deploy the Heroku Buttons for 3 modules and got it to work for Core and Dashboard but still the Storefront does not deploy. Gives the same error.

Looking for the fixes of the repo.

Thanks
